torch/csrc/autograd/record_function.h

```diff
@@ -92,14 +94,22 @@ struct TORCH_API RecordFunction {
 TORCH_API bool hasCallbacks();
 TORCH_API bool needsInputs();
 
+TORCH_API bool isSampledCallbacks();
+TORCH_API void setSampledCallbacks(bool);
+TORCH_API double getSamplingProbability();
+TORCH_API void setSamplingProbability(double);
+
 // optional argument - function's seq_no
 #define RECORD_FUNCTION(fn, inputs, ...) \
   torch::autograd::profiler::RecordFunction guard; \
   if (torch::autograd::profiler::hasCallbacks()) { \
-    if (torch::autograd::profiler::needsInputs()) { \
-      guard.before(fn, inputs, ##__VA_ARGS__); \
-    } else { \
-      guard.before(fn, ##__VA_ARGS__); \
+    if (!torch::autograd::profiler::isSampledCallbacks() || \
```

Comment:
Can you put:

```
!torch::autograd::profiler::isSampledCallbacks() || \
      guard.before(fn, inputs, ##__VA_ARGS__); \	        (((double) std::rand() / RAND_MAX) < torch::autograd::profiler::getSamplingProbability())
```

Into a (inlinable) function, so that it is easier to read what the macro is doing?

Reply:
it's a bit messy because we have to have macro definition in header but can't have global variables in header, so the usage of TORCH_API seems inevitable, though we already call TORCH_API functions like hasCallbacks or needsInputs, so it shouldn't be a problem

Reply:
or I could keep getSamplingProbability/isSampledCallbacks but I'd prefer to just have one `TORCH_API checkCallbacksEnabled` function

Reply:
I think I can't make it inline because the its body needs to be in cpp file, or I need to keep other functions